lib/parser.rb

```diff
@@ -0,0 +1,26 @@
+class Parser
+  COMMANDS = [:PLACE, :MOVE, :REPORT, :LEFT, :RIGHT]
+
+  def initialize(board)
+    @board = board
+  end
+
+  def parse(cmd)
+    cmd_arr = cmd.split(' ')
+    command = cmd_arr[0]
+    args = cmd_arr[1]
+
+    validate_command(command)
+    args ? [command.downcase] + args.split(',') : [command.downcase]
```

Comment:
elegant, but a bit complicated. divide into a few lines, the triary operator (? : ) should only be used if the three expressions are very simple, otherwise, just use if/else and divide into single command per line.

overall, I love this solution. I did it much uglier :smile: